lib/rucio/core/did_meta_plugins/__init__.py

```diff
@@ -16,186 +16,211 @@
 # Authors:
 # - Aristeidis Fkiaras <aristeidis.fkiaras@cern.ch>, 2020
 # - Benedikt Ziemons <benedikt.ziemons@cern.ch>, 2020-2021
+# - Rob Barnsley <rob.barnsley@skao.int>, 2021
 
 import importlib
 
-from rucio.common import config
-from rucio.common.exception import PolicyPackageNotFound, InvalidMetadata
+from rucio.common import config, exception
 from rucio.db.sqla.session import read_session
 
 try:
     from ConfigParser import NoOptionError, NoSectionError
 except ImportError:
     from configparser import NoOptionError, NoSectionError
 
-HARDCODED_METADATA_HANDLER_MODULE = "rucio.core.did_meta_plugins.did_column_meta.DidColumnMeta"
-FALLBACK_METADATA_HANDLER_MODULE = "rucio.core.did_meta_plugins.json_meta.JSONDidMeta"
+# Set default modules.
+#
+# - Hardcoded metadata is tied to column fields in the "dids" table. As such, types are restricted to column types and 
+#   additional fields cannot be created.
+# - Custom metadata refers to any plugin that has been written to interface with Rucio's metadata plugin system.
+# 
+# By default, the custom metadata plugin is set to the prepackaged json module. This utilises a separate table, 
+# "did_meta", allowing users to add custom metadata key/value pairs of any type.
+#
+HARDCODED_METADATA_PLUGIN_MODULE_PATH = "rucio.core.did_meta_plugins.did_column_meta.DidColumnMeta"
```

Comment:
Maybe rename this, I know it doesn't come from you, but I was never a fan of the HARDCODED and FALLBACK naming.

Reply:
How about 'base' and 'extra', or 'base' and 'custom'?

Side question: I presume it is critical to core functionality to have the metadata functions always look for metadata in the did columns first? Reason I ask is that, as it stands, there is no way to have a key with the same name as one in the did column store in a "custom" metadata store. 

It's also, in my opinion, not very transparent. If I was defining a metadata plugin in the `metadata -> plugins` section of the rucio cfg, i'd expect it to only use that plugin. That field would be better called something like `extra_plugins`.

Reply:
We mostly did it this way to preserve the functionality of how Rucio was working before (--> column information is always queried first)
But I agree, this is not very intuitive/transparent. Best would probably be to just go to an ordered list, and metadata plugins are queried in that order. The rucio default should be to do column first, but it doesn't have to be like that for every community.